homeassistant/components/reolink/number.py

```diff
@@ -515,3 +553,36 @@ async def async_set_native_value(self, value: float) -> None:
         except ReolinkError as err:
             raise HomeAssistantError(err) from err
         self.async_write_ha_state()
+
+
+class ReolinkChimeNumberEntity(ReolinkChimeCoordinatorEntity, NumberEntity):
+    """Base number entity class for Reolink IP cameras."""
+
+    entity_description: ReolinkChimeNumberEntityDescription
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        chime: Chime,
+        entity_description: ReolinkChimeNumberEntityDescription,
+    ) -> None:
+        """Initialize Reolink chime number entity."""
+        self.entity_description = entity_description
+        super().__init__(reolink_data, chime)
+
+        self._attr_mode = entity_description.mode
+
+    @property
+    def native_value(self) -> float | None:
+        """State of the number entity."""
+        return self.entity_description.value(self._chime)
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Update the current value."""
+        try:
+            await self.entity_description.method(self._chime, value)
```

Comment:
Should we type `method` to be of return type Callable?

Reply:
Method is already typed as Callable, see line 53.